// run/index.js
// const
let args = require('minimist')(process.argv.slice(2))
let generateConfigs = require('generateConfigs')
let generateDomains = require('generateDomains')
let shell = require('shelljs')
let path = require('path')

if(Object.keys(args).length == 1) args.all = true

if((args.host || args.prod || args.nobuild || args.nocertbot) && !args.notall) args.all = true
if(args.prod) args.host = 'enginx' ; 
if(args.dev) args.nocertbot = true ; args.noenginx = true
if(args.dev && !args.host) args.host = '127.0.0.1'

shell.cd(path.dirname(require.main.filename)+'/../')

let scripts = {
	volumes: `
		docker rm --force enginxvolumes
		docker run \
		--network custom \
		-v certbotconf:/things/perm/certbotconf \
		--name enginxvolumes alpine:3.9 true ;
		docker cp assets/options-ssl-nginx.conf enginxvolumes:/things/perm/certbotconf/options-ssl-nginx.conf ;
		docker cp assets/ssl-dhparams.pem enginxvolumes:/things/perm/certbotconf/ssl-dhparams.pem ;
	`,
	enginxjs: `
		${args.nobuild ? '' : 'docker build ./ -f enginxjs.Dockerfile -t enginxjs ; '}
		docker rm --force enginxjs ; 
		docker run -i \
		--network custom \
		-v sites:/enginxjs/output/tmp/sites \
		-v certbotwww:/enginxjs/output/perm/certbotwww \
		-v certbotconf:/enginxjs/output/perm/certbotconf \
		-v certbotCerts:/enginxjs/output/perm/certs/certbot \
		-v certs:/enginxjs/output/perm/certs \
		--name enginxjs enginxjs \
		node node/index.js --generateConfigs ${args.host ? '--host '+args.host : ''} --notall;
		docker rm --force enginxjs ;
	`,
	enginx: `
		${args.nobuild ? '' : 'docker build ./ -t enginx -f enginx.Dockerfile  ; '}
		docker rm --force enginx ; 
		docker run ${args.i ? '-i' : '-d'} --name enginx \
		-v sites:/things/tmp/sites \
		-v certbotwww:/things/perm/certbotwww \
		-v certbotconf:/things/perm/certbotconf \
		-v certbotCerts:/things/perm/certs/certbot \
		-v certs:/things/perm/certs \
		-p 80:80 \
		-p 8080:8080 \
		-p 443:443
		enginx

	`,
	renginx: `
		docker exec ${args.i ? '-i' : '-d'} enginx openresty -s reload
	`,
	certbotcerts: function(args){
		// docker exec -i \
		// certbot \
		return `
			${args.nobuild ? '' : 'docker build ./ -f certbot.Dockerfile -t certbot ; '}
			docker rm --force certbotcerts ; 
			docker run \
			--network custom \
			-v certbotwww:/things/perm/certbotwww \
			-v certbotconf:/etc/letsencrypt \
			-v certbotCerts:/etc/letsencrypt/live \
			--name certbotcerts \
			--entrypoint "certbot" \
			certbot \
			certonly --webroot -w /things/perm/certbotwww ${args.staging || ''} ${args.prod ? '' : '--dry-run'} --email ${args.email} ${args.domains} --rsa-key-size ${args.rsa_key_size || '4096'} --agree-tos --force-renewal  ;
		`
	},
	certbotautorenew: `
		docker rm --force certbotautorenew ;
		docker run ${args.i ? '-i' : '-d'} \
		--network custom \
		--name certbotautorenew \
		-v certbotwww:/things/perm/certbotwww \
		-v certbotconf:/etc/letsencrypt \
		-v certbotCerts:/etc/letsencrypt/live \
	--entrypoint "/autorenew.sh" \
		certbot 
	`,
	debug: `
		${args.nobuild ? '' : 'docker build ./ -f debug.Dockerfile -t enginxdebug ; '}
		docker rm --force enginxdebug ; 
		docker run ${args.i ? '-i' : '-d'} \
		--network custom \
		-v sites:/things/tmp/sites \
		-v certbotwww:/things/perm/certbotwww \
		-v certbotconf:/things/perm/certbotconf \
		-v certbotCerts:/things/perm/certs/certbot \
		-v certs:/things/perm/certs \
		--name enginxdebug enginxdebug ;
	`,
	cpdev: `

	`
}

if(args.volumes || args.all){
	console.log('running volumes')
	shell.exec(`
		${scripts.volumes}
	`)
}

if(args.enginxjs || args.all){
	console.log('running enginxjs')
	shell.exec(`
		${scripts.enginxjs}
	`)
}

if((args.enginx || args.all) && !args.noenginx){
	console.log('running enginx')
	shell.exec(`
		${scripts.enginx}
	`)
}

if(args.generateConfigs || args.all || args.dev){
	console.log('running generateConfigs')
	generateConfigs(args)
}

if(args.dev){
	console.log('running cpdev')
	shell.exec(`
		sudo mkdir -p /usr/local/etc/openresty/certs/ /usr/local/etc/openresty/conf.d/ ;
		sudo cp -a ./output/perm/certs/* /usr/local/etc/openresty/certs/ ;
		sudo cp -a ./output/tmp/sites/* /usr/local/etc/openresty/sites-enabled/ ;
		sudo cp ./assets/nginx.conf /usr/local/etc/openresty/nginx.conf ;
		sudo cp ./assets/nginx.vh.default.mac.conf /usr/local/etc/openresty/conf.d/ ;
		renginx
	`)
}

if((args.certbot || args.all) && !args.nocertbot){
	// console.log('running certbot')
	// shell.exec(`
	// 	${scripts.certbot}
	// `)
	let args2 = {
		email: 'ssl@alopu.com',
		domains: generateDomains(args)
	}
	console.log('running certbotcerts')
	shell.exec(`
		${scripts.certbotcerts({...args, ...args2})}
	`)
	console.log('running enginxjs')
	shell.exec(`
		${scripts.enginxjs}
	`)
	console.log('running renginx')
	shell.exec(`
	${scripts.renginx}
	`)
	console.log('running certbotautorenew')
	shell.exec(`
		${scripts.certbotautorenew}
	`)
}

// end
if(args.debug || args.all){
	console.log('running debug')
	shell.exec(`
		${scripts.debug}
	`)
}

module.exports = this
