// sites/node_modules/sites.js
function sites(args={}){
	return [
		{
			server_name: 'lopudesigns.com',
			certbot: true,
			dir: '/things/statics/lopudesigns.com/dist/pwa/',
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'lopu.com.au',
			certbot: true,
			dir: '/things/statics/lopu.com.au/dist/pwa/',
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'nikolajfrey.com',
			certbot: true,
			dir: '/things/statics/nikolajfrey.com/dist/pwa/',
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'beta.alopu.com',
			certbot: true,
			dir: '/things/statics/beta.alopu.com/dist/pwa/',
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'beta.api.alopu.com',
			certbot: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:9999`,
		},
		{
			server_name: 'alopu.com',
			certbot: true,
			dir: '/things/statics/alopu.com/dist/pwa/',
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'api.alopu.com',
			certbot: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:9999`,
		},
		{
			server_name: 'planetexpress.global',
			certbot: true,
			dir: '/things/statics/planetexpress.global/dist/pwa/',
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'api.planetexpress.global',
			certbot: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:9999`,
		},
		{
			server_name: 'beta.planetexpress.global',
			certbot: true,
			dir: '/things/statics/beta.planetexpress.global/dist/pwa/',
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'api.beta.planetexpress.global',
			certbot: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:9999`,
		},
		{
			server_name: 'growlights.com.au',
			certbot: true,
			dir: '/things/statics/soon.growlights.com.au/dist/pwa/',
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'beta.growlights.com.au',
			certbot: true,
			dir: '/things/statics/beta.growlights.com.au/dist/pwa/',
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'beta.api.growlights.com.au',
			certbot: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:9999`,
		},
		{
			server_name: 'api.growlights.com.au',
			certbot: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:9999`,
		},
		{
			server_name: 'growtime.com.au',
			certbot: true,
			dir: '/things/statics/soon.growtime.com.au/dist/pwa/',
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'beta.growtime.com.au',
			certbot: true,
			dir: '/things/statics/beta.growtime.com.au/dist/pwa/',
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'beta.api.growtime.com.au',
			certbot: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:9999`,
		},
		{
			server_name: 'api.growtime.com.au',
			certbot: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:9999`,
		},
		{
			server_name: 'soon.growlights.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:1338`,
		},
		{
			server_name: 'soon.dist.growlights.src',
			self_signed: true,
			dir: `/Users/lopu/things/git/lopugit/growlights/soon-growlights/dist/pwa/`,
		},
		{
			server_name: 'beta.growlights.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:1337`,
		},
		{
			server_name: 'beta.dist.growlights.src',
			self_signed: true,
			dir: `/Users/lopu/things/git/lopugit/growlights/dev-growlights/dist/pwa/`,
		},
		{
			server_name: 'growlights.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:1337`,
		},
		{
			server_name: 'dist.growlights.src',
			self_signed: true,
			dir: `/Users/lopu/things/git/lopugit/growlights/dev-growlights/dist/pwa/`,
		},
		{
			server_name: 'api.growlights.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:9999`,
		},
		{
			server_name: 'growtime.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:1337`,
		},
		{
			server_name: 'dist.growtime.src',
			self_signed: true,
			dir: `/Users/lopu/things/git/lopugit/growlights/dev-growlights/dist/pwa/`,
		},
		{
			server_name: 'api.growtime.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:9999`,
		},
		{
			server_name: 'thingopoly.src',
			self_signed: true,
		},
		{
			server_name: 'thingosphere.src',
			self_signed: true,
		},
		{
			server_name: 'beta.alopu.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:888`,
		},
		{
			server_name: 'alpha.alopu.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:8888`,
		},
		{
			server_name: 'static.alpha.alopu.src',
			self_signed: true,
			dir: '/things/statics/static.alpha.alopu.src/',,
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'static.alpha.dev.alopu.src',
			self_signed: true,
			dir: '/things/statics/static.alpha.dev.alopu.src/',,
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'static.alpha.dist.alopu.src',
			self_signed: true,
			dir: '/things/statics/static.alpha.dist.alopu.src/',,
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'static.alopu.src',
			self_signed: true,
			dir: '/things/statics/static.alopu.src/',,
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'static.beta.dev.alopu.src',
			self_signed: true,
			dir: '/things/statics/static.beta.dev.alopu.src/',,
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'static.beta.dist.alopu.src',
			self_signed: true,
			dir: '/things/statics/static.beta.dist.alopu.src/',,
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'static.dev.alopu.src',
			self_signed: true,
			dir: '/things/statics/static.dev.alopu.src/',,
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'static.dist.alopu.src',
			self_signed: true,
			dir: '/things/statics/static.dist.alopu.src/',,
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'alopu.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:8888`,
		},
		{
			server_name: 'api.alopu.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:9999`,
		},
		{
			server_name: 'lopudesigns.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:7777`,
		},
		{
			server_name: 'auth.alopu.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:9999`,
		},
		{
			server_name: 'planetexpress.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:7788`,
		},
		{
			server_name: 'static.planetexpress.src',
			self_signed: true,
			dir: '/things/statics/static.planetexpress.src/',,
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'static.dev.planetexpress.src',
			self_signed: true,
			dir: '/things/statics/static.dev.planetexpress.src/',,
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'static.dist.planetexpress.src',
			self_signed: true,
			dir: '/things/statics/static.dist.planetexpress.src/',,
			rewrite: 'rewrite ^.*$ /;'
		},
		{
			server_name: 'dingern.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:9876`,
		},
		{
			server_name: 'api.dingern.src',
			self_signed: true,
			ip: `http://${args.host || 'enginx' || 'host.docker.internal'}:9999`,
		},
		{
			server_name: 'static.dingern.src',
			self_signed: true,
			dir: '/things/statics/static.dingern.src/',,
			rewrite: 'rewrite ^.*$ /;'
		},
	]
}

module.exports = sites
